src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -458,42 +473,53 @@ private bool HandleError(C4Error error)
             return true;
         }
 
-        private void OnDocEnded(List<ReplicatedDocument> replications, bool pushing)
+        private void OnDocEnded(ConcurrentBag<ReplicatedDocument> replications, bool pushing)
         {
             if (_disposed) {
                 return;
             }
 
-            for(int i = 0; i < replications.Count; i++) {
+            while (!replications.IsEmpty) {
```

Comment:
This is not a good loop.  It will keep looping adding in tasks that will eventually do nothing at the end because its termination condition is delayed.